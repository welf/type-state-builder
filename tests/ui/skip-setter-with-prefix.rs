use type_state_builder::TypeStateBuilder;

#[derive(TypeStateBuilder)]
struct InvalidStruct {
    name: String,
    
    // This should be an error: can't have setter prefix on skipped setters
    #[builder(default = "42", skip_setter, setter_prefix = "with_")]
    count: u32,
}

fn main() {
    let _instance = InvalidStruct::builder()
        .name("test".to_string())
        .build();
}