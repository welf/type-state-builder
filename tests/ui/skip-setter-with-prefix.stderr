error: Field-level setter_prefix is incompatible with skip_setter. Remove one of these attributes.
 --> tests/ui/skip-setter-with-prefix.rs:3:10
  |
3 | #[derive(TypeStateBuilder)]
  |          ^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the derive macro `TypeStateBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no function or associated item named `builder` found for struct `InvalidStruct` in the current scope
  --> tests/ui/skip-setter-with-prefix.rs:13:36
   |
4  | struct InvalidStruct {
   | -------------------- function or associated item `builder` not found for this struct
...
13 |     let _instance = InvalidStruct::builder()
   |                                    ^^^^^^^ function or associated item not found in `InvalidStruct`
